build-system/tasks/prettify.js

```diff
@@ -0,0 +1,144 @@
+ * which uses eslint.)
+ */
+'use strict';
+
+const argv = require('minimist')(process.argv.slice(2));
+const gulp = require('gulp');
+const log = require('fancy-log');
+const path = require('path');
+const tap = require('gulp-tap');
+const {getOutput} = require('../common/exec');
+const {green, cyan, red, yellow} = require('ansi-colors');
+const {highlight} = require('cli-highlight');
+const {isTravisBuild} = require('../common/travis');
+const {logOnSameLine} = require('../common/utils');
+const {prettifyGlobs} = require('../test-configs/config');
+
+const rootDir = path.dirname(path.dirname(__dirname));
+const prettierCmd = 'node_modules/.bin/prettier';
+
+/**
+ * Prettier exit codes. See https://prettier.io/docs/en/cli.html#exit-codes
+ * @enum {number}
+ */
+const PrettierResult = {
+  SUCCESS: 0,
+  FAILURE: 1,
+  ERROR: 2,
+};
+
+/**
+ * Checks files for formatting (and optionally fixes them) with Prettier.
+ * Returns the cumulative result to the `gulp` process via  process.exitCode so
+ * that all files can be checked / fixed.
+ *
+ * @return {!Promise}
+ */
+function prettify() {
+  const filesToCheck = argv.files ? argv.files.split(',') : prettifyGlobs;
+  return gulp
+    .src(filesToCheck)
+    .pipe(
+      tap(file => {
+        checkFile(path.relative(rootDir, file.path));
+      })
+    )
+    .on('finish', () => {
+      if (process.exitCode == 1) {
+        logOnSameLine(red('ERROR: ') + 'Found errors in one or more files.');
+        if (!argv.fix) {
+          log(
+            yellow('NOTE 1:'),
+            'You may be able to automatically fix some of these warnings ' +
+              '/ errors by running',
+            cyan('gulp prettify --fix'),
+            'from your local branch.'
+          );
+          log(
+            yellow('NOTE 2:'),
+            'Since this is a destructive operation (that edits your files',
+            'in-place), make sure you commit before running the command.'
+          );
+        }
+      } else {
+        if (!isTravisBuild()) {
+          logOnSameLine(green('SUCCESS: ') + 'No formatting errors found.');
+        }
+      }
+    });
+}
+
+/**
+ * Fixes the formatting of a single file
+ * @param {string} file
+ */
+function fixFile(file) {
+  const fixCmd = `${prettierCmd} --write ${file}`;
+  const fixResult = getOutput(fixCmd);
+  if (fixResult.status == PrettierResult.SUCCESS) {
+    logOnSameLine(green('Fixed: ') + file + '\n');
+  } else {
+    logOnSameLine(red('Could not fix: ') + file);
+    console.log(highlight(fixResult.stderr, {ignoreIllegals: true}), '\n');
+    process.exitCode = 1;
+  }
+}
+
+/**
+ * Checks and optionally fixes the formatting of a single file.
+ * @param {string} file
+ */
+function checkFile(file) {
+  const checkCmd = `${prettierCmd} --list-different ${file}`;
+  const checkResult = getOutput(checkCmd);
+  if (checkResult.status == PrettierResult.SUCCESS) {
+    if (!isTravisBuild()) {
```

Comment:
Doesn't `logOnSameLine` already check `!isTravisBuild()`?

Reply:
That condition is different. (It also checks for `process.stdout.isTTY`).

I'm sure we can clean up many of the `isTravisBuild()` calls across `build-system`, but it'll have to be in a separate PR.